batchflow/models/torch/base.py

```diff
@@ -1343,3 +1227,349 @@ def load(self, path, *args, eval=False, **kwargs):
 
         if eval:
             self.model.eval()
+
+
+    # Debug and profile the performance
+    def set_debug_mode(self, mode=True):
+        """ Changes representation of model to a more or less detailed.
+        By default, model representation reduces the description of the most complex modules.
+        """
+        if self.model is None:
+            raise ValueError('Model is not initialized yet. ')
+        self.model.apply(lambda module: setattr(module, 'debug', mode))
+
+    def show_profile_info(self, per_iter=False, sortby=None, limit=10, parse=False):
+        """ Show stored profiling information with varying levels of details. """
+        if (self.profile_info is None) or parse:
+            self._parse_profilers()
+
+        if self.device.type == 'cpu':
+            columns = ['ncalls', 'CPU_tottime', 'CPU_cumtime', 'CPU_tottime_avg']
+            if sortby is None:
+                sortby = ('CPU_tottime', 'sum') if per_iter is False else 'CPU_tottime'
+        else:
+            columns = ['ncalls', 'CUDA_cumtime', 'CUDA_cumtime_avg']
+            if sortby is None:
+                sortby = ('CUDA_cumtime', 'sum') if per_iter is False else 'CUDA_cumtime'
+
+        if per_iter is False:
+            aggs = {key: ['sum', 'mean', 'max'] for key in columns}
+            result = (self.profile_info.reset_index().groupby(['name']).agg(aggs)
+                      .sort_values(sortby, ascending=False)[:limit])
+        else:
+            result = (self.profile_info.reset_index().set_index(['iter', 'name'])[columns]
+                      .sort_values(['iter', sortby], ascending=[True, False])
+                      .groupby(level=0).apply(lambda df: df[:limit]).droplevel(0))
+        return result
+
+    def _parse_profilers(self):
+        us_in_s = 1000.0 * 1000.0
+
+        indices, values = [], []
+        for i, profiler in enumerate(self.profilers):
+            for evt in profiler.function_events.key_averages():
+                indices.append((i, evt.key))
+                row_dict = {
+                    'ncalls': evt.count,
+                    'CPU_tottime': evt.self_cpu_time_total / us_in_s,
+                    'CPU_cumtime': evt.cpu_time_total / us_in_s,
+                    'CUDA_cumtime': evt.cuda_time_total / us_in_s,
+                }
+                values.append(row_dict)
+        multiindex = pd.MultiIndex.from_tuples(indices, names=['iter', 'name'])
+
+        self.profile_info = pd.DataFrame(values, index=multiindex,
+                                         columns=['ncalls', 'CPU_tottime', 'CPU_cumtime', 'CUDA_cumtime'])
+        self.profile_info['CPU_tottime_avg'] = self.profile_info['CPU_tottime'] / self.profile_info['ncalls']
+        self.profile_info['CUDA_cumtime_avg'] = self.profile_info['CUDA_cumtime'] / self.profile_info['ncalls']
+
+
+    # Textual visualization of model
+    def information(self, config=True, devices=True, train_steps=True, model=False, misc=False):
```

Comment:
Make it as clean as possible. It should gather info into a readable structure and print only if needed (or no print at all)